bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/scalarweb/models/api/CurrentExternalTerminalsStatus_1_0.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.sony.internal.scalarweb.models.api;
+
+import java.util.Arrays;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang.Validate;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * This class represents a current external terminal status
+ *
+ * @author Tim Roberts - Initial contribution
+ */
+@NonNullByDefault
+public class CurrentExternalTerminalsStatus_1_0 {
+
+    public static final String META_ZONEOUTPUT = "meta:zone:output";
+
+    /** The status of the terminal */
+    private @Nullable String active;
+
+    /** The connection status */
+    private @Nullable String connection;
+
+    /** The icon for the terminal */
+    private @Nullable String iconUrl;
+
+    /** The label of the terminal */
+    private @Nullable String label;
+
+    /** The meta identifying information on the terminal */
+    private final @Nullable String meta;
+
+    /** The outputs associated with this terminal (usually an input) */
+    private @Nullable String @Nullable [] outputs;
+
+    /** The title (name) of the terminal */
+    private final @Nullable String title;
+
+    /** The uri identifying the terminal */
+    private final @Nullable String uri;
+
+    public CurrentExternalTerminalsStatus_1_0(final String uri, final String title) {
+        this.uri = uri;
+        this.title = title;
+        this.meta = META_ZONEOUTPUT;
+    }
```

Comment:
This is the only constructor for this class. There is no default constructor for a deserializer to use. There are no setter fields, and most of the fields are not set by the constructor. Am I missing something? How is this class even used?

Reply:
gson has never needed a no-arg constructor and will ignore arg constructors during deserialization.  So deserialization will construct the class directly - set the fields and the code retrieves them from the getters.  For serialization, my code calls the constructor (which sets the fields) and it's serialized.  No need for setters.

I just googled and you can read about it here:  https://stackoverflow.com/questions/18645050/is-default-no-args-constructor-mandatory-for-gson

I may take this review item as needing to add no-arg constructors simply to follow their recommendation (however, there are soo many classes I'm a bit on the fence about doing this)

Reply:
I didn't know that gson is doing magic behind scenes. Guess I learned something. But yea, even gson recommends having a no-arg constructor.